### ISSUE TITLE
util-linux: Depends on ncurses: lib/colors.c:14:21: fatal error: curses.h: No such file or directory

### ISSUE THREAD

**Issue by joshuap** [Linuxbrew/homebrew-core#1175]
When doing `brew install wget` (fairly fresh Ubuntu Gnome/Linuxbrew install) I was getting the following error. The solution was to `brew install ncurses` first:

```
~ » brew install wget
==> Installing dependencies for wget: util-linux
==> Installing wget dependency: util-linux
==> Downloading https://www.kernel.org/pub/linux/utils/util-linux/v2.27/util-linux-2.27.1.tar.xz
Already downloaded: /home/josh/.cache/Homebrew/util-linux-2.27.1.tar.xz
==> ./configure --disable-silent-rules --prefix=/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1 --disable-use-tty-group --disable-kill --disable-cal --disable-ul
==> make install
Last 15 lines from /home/josh/.cache/Homebrew/Logs/util-linux/02.make:
/bin/bash ./libtool  --tag=CC   --mode=compile /usr/bin/gcc-5 -DHAVE_CONFIG_H -I.  -include config.h -I./include -DLOCALEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/share/locale\" -D_PATH_LOCALSTATEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/var\" -isystem/home/josh/.linuxbrew/include -fsigned-char -fno-common -Wall -Werror=sequence-point -Wextra -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wno-missing-field-initializers -Wredundant-decls -Wsign-compare -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wnested-externs -Wpointer-arith -Wstrict-prototypes -Wformat-security -Wimplicit-function-declaration  -Os -w -pipe -march=native -c -o lib/libtcolors_la-colors.lo `test -f 'lib/colors.c' || echo './'`lib/colors.c
libtool: compile:  /usr/bin/gcc-5 -DHAVE_CONFIG_H -I. -include config.h -I./include -DLOCALEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/share/locale\" -D_PATH_LOCALSTATEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/var\" -isystem/home/josh/.linuxbrew/include -fsigned-char -fno-common -Wall -Werror=sequence-point -Wextra -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wno-missing-field-initializers -Wredundant-decls -Wsign-compare -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wnested-externs -Wpointer-arith -Wstrict-prototypes -Wformat-security -Wimplicit-function-declaration -Os -w -pipe -march=native -c lib/colors.c  -fPIC -DPIC -o lib/.libs/libtcolors_la-colors.o
lib/colors.c:14:21: fatal error: curses.h: No such file or directory
compilation terminated.
Makefile:8580: recipe for target 'lib/libtcolors_la-colors.lo' failed
make[2]: *** [lib/libtcolors_la-colors.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  /usr/bin/gcc-5 -DHAVE_CONFIG_H -I. -include config.h -I./include -DLOCALEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/share/locale\" -D_PATH_LOCALSTATEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/var\" -isystem/home/josh/.linuxbrew/include -fsigned-char -fno-common -Wall -Werror=sequence-point -Wextra -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wno-missing-field-initializers -Wredundant-decls -Wsign-compare -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wnested-externs -Wpointer-arith -Wstrict-prototypes -Wformat-security -Wimplicit-function-declaration -Os -w -pipe -march=native -c lib/cpuset.c -o lib/libcommon_la-cpuset.o >/dev/null 2>&1
libtool: compile:  /usr/bin/gcc-5 -DHAVE_CONFIG_H -I. -include config.h -I./include -DLOCALEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/share/locale\" -D_PATH_LOCALSTATEDIR=\"/home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/var\" -isystem/home/josh/.linuxbrew/include -fsigned-char -fno-common -Wall -Werror=sequence-point -Wextra -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wno-missing-field-initializers -Wredundant-decls -Wsign-compare -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wnested-externs -Wpointer-arith -Wstrict-prototypes -Wformat-security -Wimplicit-function-declaration -Os -w -pipe -march=native -c lib/loopdev.c -o lib/libcommon_la-loopdev.o >/dev/null 2>&1
make[2]: Leaving directory '/tmp/util-linux-20161031-16274-1vsnl31/util-linux-2.27.1'
Makefile:11208: recipe for target 'install-recursive' failed
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory '/tmp/util-linux-20161031-16274-1vsnl31/util-linux-2.27.1'
Makefile:11666: recipe for target 'install' failed
make: *** [install] Error 2
```

**Comment by sjackman** [comment]
Thanks for the bug report, Joshua. Please submit a PR if you'd like. Run `brew edit wget` and add
```ruby
depends_on "homebrew/dupes/ncurses" unless OS.mac?
```

**Comment by sjackman** [comment]
We were able to build a bottle for `wget` without this `ncurses` dependency, so it's odd that your system requires it but the CI system does not. Please report the output of
`brew linkage wget`. For example:
```
❯❯❯ brew linkage wget
Homebrew libraries:
  /gsc/btl/linuxbrew/lib/libc.so.6 (glibc)
  /gsc/btl/linuxbrew/lib/libdl.so.2 (glibc)
  /gsc/btl/linuxbrew/lib/libcrypto.so.1.0.0 (openssl)
  /gsc/btl/linuxbrew/lib/libssl.so.1.0.0 (openssl)
  /gsc/btl/linuxbrew/lib/libuuid.so.1 (util-linux)
  /gsc/btl/linuxbrew/lib/libz.so.1 (zlib)
Possible undeclared dependencies:
  zlib
```

**Comment by joshuap** [comment]
Here you go:

```
System libraries:
  
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libdl.so.2
  /lib64/ld-linux-x86-64.so.2
Homebrew libraries:
  /home/josh/.linuxbrew/lib/libcrypto.so.1.0.0 (openssl)
  /home/josh/.linuxbrew/lib/libssl.so.1.0.0 (openssl)
  /home/josh/.linuxbrew/lib/libuuid.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libz.so.1 (zlib)
Possible undeclared dependencies:
  zlib
```

**Comment by sjackman** [comment]
Also no apparent `ncurses` dependency. Oh, I misread the log above. It's `util-linux` that has the dependency on `ncurses`, not `wget`.
```
$ brew linkage util-linux
Homebrew libraries:
…
Possible undeclared dependencies:
  homebrew/dupes/ncurses
  python
  zlib
```

**Comment by sjackman** [comment]
I've opened PR #1176 to resolve this issue.

**Comment by joshuap** [comment]
Oh yeah, sorry about that. Here's my linkage for util-linux in case you still need it:

```
System libraries:
  
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libcrypt.so.1
  /lib/x86_64-linux-gnu/libpthread.so.0
  /lib/x86_64-linux-gnu/libreadline.so.6
  /lib/x86_64-linux-gnu/librt.so.1
  /lib/x86_64-linux-gnu/libtinfo.so.5
  /lib/x86_64-linux-gnu/libutil.so.1
  /lib64/ld-linux-x86-64.so.2
Homebrew libraries:
  /home/josh/.linuxbrew/lib/libncursesw.so.6 (homebrew/dupes/ncurses)
  /home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/lib/libblkid.so.1 (util-linux)
  /home/josh/.linuxbrew/Cellar/util-linux/2.27.1_1/lib/libuuid.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libblkid.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libfdisk.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libmount.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libsmartcols.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libuuid.so.1 (util-linux)
  /home/josh/.linuxbrew/lib/libz.so.1 (zlib)
Possible undeclared dependencies:
  homebrew/dupes/ncurses
  zlib
```

Thanks for the quick responses!

**Comment by sjackman** [comment]
No worries. Thanks for the bug report! I've merged #1176.
